samza-core/src/main/java/org/apache/samza/config/ClusterManagerConfig.java

```diff
@@ -86,7 +86,7 @@
   private static final int DEFAULT_CONTAINER_RETRY_COUNT = 8;
 
   /**
-   * Determines whether a JMX server should be started on the job coordinator
+   * Determines whether a JMX server should be started on the job coordinator and container
```

Comment:
These are the following ways:

- If we don't go with current config we Introduce a single config `job.jmx.enabled` for standalone and yarn. We will have to Deprecate configs like `cluster-manager.jobcoordinator.jmx.enabled` and `yarn.am.jmx.enabled`

- We work this out with two different configs, then container needs to refactor and do that check for standalone vs non-standalone deployement (this makes he code even messier)

- We go with the `cluster-manager.jobcoordinator.jmx.enabled` and document that this applies for standalone, this does not need deprecation

Is there a namespace for stanadalone only?

Reply:
Existing implementation of `ClusterManagerConfig.getJmxEnabled` checks the following two configuration keys: `cluster-manager.jobcoordinator.jmx.enabled`, `yarn.am.jmx.enabled` and returns true if either of them are true. If we use that config method for standalone , users can turn off/on the jmxServer via `yarn.am.jmx.enabled`. 

I think it'll be wise not to use yarn configuration for standalone  and simply document it. I think both `cluster-manager`, `yarn` namespace doesn't apply for standalone.

>> Is there a namespace for standalone only ? 
Yes, `job.coordinator.`  is predominantly used for most of the standalone configuration(or one can use `job.jmx.enabled` for this case or some other meaningful name). 

@prateekm what do you think?

Reply:
IIUC, there is a config to enable/disable JMX on the AM, but not one for the container, where it's always on. 
If so, we can add `job.jmx.enabled` with defaut=true, and use it consistently for both AM and containers. Should respect the values of existing configs. I.e., they should have higher precedence than job.jmx.enabled. Should mark both existing configs as deprecated in the configuration reference.

Reply:
I propose a better plan (already did according to first commit)

That's what I did in the first commit, introduced a new config called `cluster-manager.container.jmx.enabled`, maybe I will rename it to `job.container.jmx.enabled`

and let the current config `cluster-manager.jobcoordinator.jmx.enabled`  manage JMX on JobCoordinator itself, this is more transparent in terms of handling JMX differently in JonCoordinator and Container since they apply differently in different ways of deploying a job

Reply:
This way we eventually end up with 2 configs, the other way we end up with one. Unless there's a reason to have separate configs for the AM and the containers, fewer configs seems better to me. WDYT?

Reply:
I aggree